tests/testthat/test-table_with_settings_ui.R

```diff
@@ -0,0 +1,92 @@
+library(shinytest2)
+
+testthat::test_that("maxi version", {
+  skip_on_cran()
+  skip_on_ci()
+  app <- AppDriver$new(
+    app_tws(),
+    name = "tws",
+    variant = "app_tws_max",
+    )
+  # check standard file naming
+  file_name <- app$get_value(input = "table_with_settings-downbutton-file_name")
+  testthat::expect_equal(
+    sub("_\\d{6}$", "", file_name),
+    paste0("table_", strftime(Sys.Date(), format = "%Y%m%d"))
+  )
+  app$expect_text("table")
+  app$set_window_size(width = 1392, height = 763)
+
+  # save txt (default)
+  app$click("table_with_settings-downbutton-dwnl")
+  app$set_inputs(`table_with_settings-downbutton-dwnl_state` = TRUE)
+  app$set_inputs(`table_with_settings-downbutton-file_name` = "table1")
+  app$expect_download("table_with_settings-downbutton-data_download")
```

Comment:
This seems to be tested [here](https://github.com/insightsengineering/teal.widgets/blob/1012a01175f9c12382690899715f5b9f13dbd41a/tests/testthat/test-table_with_settings.R#L68) I'm not sure if we need to test it again. We can think whether we want to test snapshots of this

Reply:
There we only check whether the file to be downloaded exists and changed it name properly, here we also check the exact content of the tables (txt and csv, not pdf).